### DISCUSSION TITLE
ideas to consider please

### DISCUSSION THREAD

**Discussion by 098tomtom** [simpleavr/s14clock#5] [Ideas]
Love the display.

Any thoughts on being able to retrieve and display tweets or telegram messages, or somehow send a message to display without using the current interface/being on the same network?

Any room for daisy chaining several displays together and having text scroll from one unit to the next?

Thanks

**Comment by simpleavr** [comment]
Thanks for your interest.

For using the S14 Clock as a messaging display, apart from the message box in the web interface, one can use HTTP GET to place a message as part of a URL, similar to a RESTFUL interface.

So instead of specializing the firmware for tweets (or for weather info, stock info, bus route, etc), I intend to just provide a better interface so that S14 can be integrated as part of other specialized projects.

For example, if you want to display "NOW SERVING CUST 223" (in a restaurant usage), you can use "curl" in command line interface like so;

`curl 192.168.2.29/msg/NOW_SERVING_NO_223`

This can also be done in a script if you are to automate the usage. I am not familiar w/ capturing tweets. And **you can integrate tweeter clients** (perhaps like this one [https://github.com/marquisvictor/twint](url)) in a script to capture and forward messages.

Currently this scheme (using /msg/ node) cannot show spaces, I will fix that in the next release. Also you can prefix your message with "~X" or "~x" so that your message will disappear in 60 or 10 seconds. And the display will return to whatever it was displaying.

To open up the messaging interface to the internet, you will need to find the common solution of **setting up "port forwarding" to the S14 unit**. This is out of scope of the project and really becomes a general home network configuration issue. You can google the topic and experiment with it.

As for daisy chaining additional display units, this is something I am considering in the longer term. The challenge is that I had used up all the IO pins (except for one) of the S2 Mini and it will be quite difficult to implement. Even if I have it implemented, there will be fewer or no "transition effects" as these effect requires complete and timely control of individual digits / character positions. If it will be implements, only simple scrolling will be provided. I will however, put it in the list as I am also interested in the hardware design of it.
